### PR TITLE
Make the configuration object a class.

### PR DESCRIPTION
**Pull request by danpovey**

### CODE REVIEW

**Comment by danpovey** [comment]
This is a pull request to make the core configuration object a class.
The default config looks like this:
```
num_classes 2
num_colors 1
padding 10
train_image_size 256
offsets 1 0  0 1  -2 -1  1 -2  3 2  -4 3  -4 -7  10 -4  3 15  -21 0
```
Notice that I added some very basic test code.  When people work on other files it would be cool if they could add similar test code, to at least make sure things run.  (but don't let this delay you, it's not a must).   It might be helpful to have functions to create random objects of the various types we use, for testing purposes.

File: scripts/waldo/core_config.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        # We can change the assertion that num_colors <= 3 later on if we ever
+        # need to operate on images with larger color spaces.
+        assert self.num_colors >= 1 and self.num_colors <= 3
+
+        assert len(self.offsets) > 0 and not (0,0) in self.offsets
+        offsets_set = set(self.offsets)
+        assert len(offsets_set) == len(self.offsets)
+        for o in self.offsets:
+            assert isinstance(o, tuple) and len(o) == 2
+            assert 0 != (0,0)
```
Comment (hhadian) [inline scripts/waldo/core_config.py]: I guess this line can be removed.
BTW, I think this class is much better than a dict; I'll update the segmenter code to use this once Yiwen updates the training code.

Comment (danpovey) [reply scripts/waldo/core_config.py]: Oh, what was supposed to be "o" not "0".  Would you mind changing it next time you push something?

Comment (hhadian) [reply scripts/waldo/core_config.py]: Yes, I noticed but the same assertion is also done in L64. Will do it in the next PR I make (tonight).

**Pull request merged**